### PR TITLE
lightning: update dependency pebbledb to v1.1.0

### LINKED ISSUE #52402: Update dependency pebbledb to an official release version
## Enhancement

[pebble@v1.1.0](https://github.com/cockroachdb/pebble/tree/v1.1.0) should be more stable than the current used commit hash. I want to update to dependency to the version because there are some known issues in TiCDC:

* https://github.com/pingcap/tiflow/issues/6367
* https://github.com/pingcap/tiflow/issues/10853

So let's upgrade it in both TiDB and TiCDC.

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Update dependency pebbledb to v1.1.0.

Issue Number: close #52402 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  Only a dependency is updated.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by 3pointer — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by D3Hunter** [comment]
https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_lightning_integration_test/detail/pull_lightning_integration_test/159/pipeline
failed

**Comment by D3Hunter** [comment]
ci pass https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_lightning_integration_test/detail/pull_lightning_integration_test/166/pipeline

File: pkg/lightning/backend/local/engine.go [hunk 1]
```diff
@@ -1393,9 +1402,8 @@ func (sw *sstWriter) writeKVs(kvs []common.KvPair) error {
 	internalKey := sstable.InternalKey{
 		Trailer: uint64(sstable.InternalKeyKindSet),
 	}
-	var lastKey []byte
 	for _, p := range kvs {
-		if bytes.Equal(p.Key, lastKey) {
+		if sw.lastKey != nil && bytes.Equal(p.Key, sw.lastKey) {
```
Comment (D3Hunter) [inline pkg/lightning/backend/local/engine.go:1406]: why we cannot use local `lastKey` as before, what's changed?

File: pkg/util/extsort/disk_sorter.go [hunk 2]
```diff
@@ -1378,7 +1392,12 @@ func (w *diskSorterWriter) flush() error {
 	slices.SortFunc(w.kvs, func(a, b keyValue) int {
 		return bytes.Compare(a.key, b.key)
 	})
+	var lastKey []byte
 	for _, kv := range w.kvs {
+		if bytes.Equal(lastKey, kv.key) {
+			continue
```
Comment (D3Hunter) [inline pkg/util/extsort/disk_sorter.go:1398]: why we have to check lastKey now?

File: pkg/lightning/backend/local/engine.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/lightning/backend/local/engine.go:1406]: maybe add some comment to explain why, as we don't need it in previous version

Comment (hicqu) [reply pkg/lightning/backend/local/engine.go:1410]: `writeKVs` is allowed to be called more than one times. However, in one SST file all keys should be sorted and deduped. So we can't use a local variable in `writeKVs`.

File: pkg/util/extsort/disk_sorter.go (hunk 2 above)
Comment (hicqu) [reply pkg/util/extsort/disk_sorter.go:1401]: The stable pebbledb requires all keys are sorted and deduped when construct a SST file.

**Review by D3Hunter — APPROVED** [review]

**Comment by wuhuizuo** [comment]
> /hold

GraphQL: Merge already in progress (mergePullRequest)

broken by https://www.githubstatus.com/incidents/k7bc8pkxpmbp

**Pull request merged**